icdiff

```diff
@@ -748,6 +764,9 @@ def diff_files(options, a, b):
         lines_b = [line_b for line_b in lines_b if
                    not re.search(options.matcher, line_b)]
 
+    # Determine if a difference has been detected
+    diff_found = len(lines_a) | len(lines_b)
```

Comment:
This was a bug: this should have been `lines_a != lines_b`.  Fixed in 9526c219bd0bf994924f087c9f87f4ae82a44039